app/services/identity/create.rb

```diff
@@ -20,7 +20,10 @@ def initialize(user:, origin: :ecf, email: nil)
       @email = email || user.email
     end
 
-    def find_or_create_identity
+    def find_or_create_identity!
+      email_user = Identity.find_user_by(email:)
+      raise "Email already taken" if email_user.present? && email_user != user
```

Comment:
Not one for now but we do this in a few places, could be worth creating a service class

Reply:
Yes good idea, I noticed this seemed to be missing so shoved it in with this PR 👍